### PR TITLE
CVE-2007-0404 and CVE-2009-2693

### PR DESCRIPTION
**Pull request by ADAdebomi**
Did research on CVE-2007-0404 and populated YAML file accordingly

### CODE REVIEW

File: cves/CVE-2007-0404.yml [hunk 1]
```diff
@@ -49,27 +49,39 @@ description_instructions: |
 
   Your target audience is people just like you before you took any course in
   security
-description:
+description: A script for compliling messages was made vulnerable because the inputs
+  submitted were not being checked to make sure that they were not malicious in nature.
+  The script job was to make OS level commands and if the inputs were not checked correctly
+  changes could to be made to the system if a targetted input is submitted as an arguememnt or in the .po file
+  with commands. If some of these commands are exectuted hackers can take over the system by giving themselves
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: Typo: executed

Comment (dwelcheck) [reply cves/CVE-2007-0404.yml]: You need a comma here: _If some of these commands are executed,_

File: cves/CVE-2007-0404.yml [hunk 2]
```diff
@@ -49,27 +49,39 @@ description_instructions: |
 
   Your target audience is people just like you before you took any course in
   security
-description:
+description: A script for compliling messages was made vulnerable because the inputs
+  submitted were not being checked to make sure that they were not malicious in nature.
+  The script job was to make OS level commands and if the inputs were not checked correctly
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: Did you mean "The script's job"?

Comment (dwelcheck) [reply cves/CVE-2007-0404.yml]: OS-level commands

File: cves/CVE-2007-0404.yml [hunk 3]
```diff
@@ -49,27 +49,39 @@ description_instructions: |
...
+  changes could to be made to the system if a targetted input is submitted as an arguememnt or in the .po file
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: Typo: argument

File: cves/CVE-2007-0404.yml (hunk 2 above)
Comment (dwelcheck) [reply cves/CVE-2007-0404.yml]: This sentence is trying to say a bit too much and becomes hard to read after the second or third clause.  Try breaking it up into 2-3 sentences.

File: cves/CVE-2007-0404.yml [hunk 4]
```diff
@@ -49,27 +49,39 @@ description_instructions: |
...
   was wrong. Otherwise, leave it blank.
-bounty:
+bounty: 
   amt:
   announced:
   url:
 reviews: []
-bugs: []
-repo:
+bugs: [407519]
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: I believe this bug # is from Debian, not from Django itself.

File: cves/CVE-2007-0404.yml [hunk 5]
```diff
@@ -49,27 +49,39 @@ description_instructions: |
...
-  note:
+  note: The os shell line added is used to parse the command line and strips
+   it of backticks, dollar sign usage and any command substitutions and parameter expansions
+   that may run commands that make the system vulnerable
+
 - commit: a132d411c6986418ee6c0edc331080aa792fee6e
-  note:
-vccs: []
+  note: Handles submitting and applying security fix for the changes made in commit 
+    518d406e53a7417385a1a2e10bb5110b67d28fac in a windowns computer set up
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: typo: windows

Comment (dwelcheck) [reply cves/CVE-2007-0404.yml]: I would get rid of "set up", it's not necessary to understand what the sentence is saying

File: cves/CVE-2007-0404.yml [hunk 6]
```diff
@@ -49,27 +49,39 @@ description_instructions: |
...
+vccs: [5cf8f684237]
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: Could you change this to follow the format of the `fixes` block?

File: cves/CVE-2007-0404.yml [hunk 7]
```diff
@@ -108,10 +120,13 @@ discovered:
     The automated, contest, and developer flags can be true, false, or nil.
 
     If there is no evidence as to how this vulnerability was found, then please explain where you looked.
-  answer:
-  automated:
-  contest:
-  developer:
+  answer: It was discovered when the script that was created using that portion of code was not escaping 
+    files after it ran. This led to the developers realizing that it could also lead to malicious scripts
+    to be ran since the commands werent filtered and sanitized. It was also noticed by another developer
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: typo: weren't

File: cves/CVE-2007-0404.yml [hunk 8]
```diff
@@ -126,8 +141,10 @@ autodiscoverable:
 
     The answer field should be boolean. In answer_note, please explain
     why you come to that conclusion.
-  answer_note:
-  answer:
+  answer_note: The reason this vulnerability occured was because the code writer did not test or check or
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: I would change "code writer" to "developer"

Comment (dwelcheck) [reply cves/CVE-2007-0404.yml]: I would try to reduce the redundancy of "did not test or check or take security seriously enough"

File: cves/CVE-2007-0404.yml [hunk 9]
```diff
@@ -192,8 +211,8 @@ ipc:
     software system reads is another form of IPC.
 
     Answer should be boolean.
-  answer:
-  note:
+  answer: false
+  note: None
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: Leave this entirely blank rather than having `None` if the answer is false.

File: cves/CVE-2007-0404.yml [hunk 10]
```diff
@@ -254,7 +273,13 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: First and foremost, it doesnt look like any test was handled on this by attempting
```
Comment (dwelcheck) [inline cves/CVE-2007-0404.yml]: typo: doesn't

Comment (dwelcheck) [reply cves/CVE-2007-0404.yml]: Saying "any test handled this case" would make the sentence flow better.

File: cves/CVE-2007-0404.yml (hunk 1 above)
Comment (ADAdebomi) [reply cves/CVE-2007-0404.yml]: Thanks will fix these

File: cves/CVE-2007-0404.yml (hunk 3 above)
Comment (ADAdebomi) [reply cves/CVE-2007-0404.yml]: Will fix, thanks

File: cves/CVE-2007-0404.yml (hunk 4 above)
Comment (ADAdebomi) [reply cves/CVE-2007-0404.yml]: Yes, you're correct, I couldn't find a bug on the Django end as it was part of a release and didn't point to any bug fix. Will fix accordingly

**Comment by dwelcheck** [comment]
+4 upvotes

**Pull request merged**